File: Discord_android/Discord/app/src/main/java/com/peanut/discord/models/ChannelPermission.java
package com.peanut.discord.models;
public class ChannelPermission {
    private int channelId;
    private int roleId;
    private boolean viewMessage;
    private boolean react;
    private boolean sendMessage;
    private boolean sendImage;
    private Channel channel;
    private Role role;
    public int getChannelId() {
        return channelId;
    }
    public void setChannelId(int channelId) {
        this.channelId = channelId;
    }
    public int getRoleId() {
        return roleId;
    }
    public void setRoleId(int roleId) {
        this.roleId = roleId;
    }
    public boolean isViewMessage() {
        return viewMessage;
    }
    public void setViewMessage(boolean viewMessage) {
        this.viewMessage = viewMessage;
    }
    public boolean isReact() {
        return react;
    }
    public void setReact(boolean react) {
        this.react = react;
    }
    public boolean isSendMessage() {
        return sendMessage;
    }
    public void setSendMessage(boolean sendMessage) {
        this.sendMessage = sendMessage;
    }
    public boolean isSendImage() {
        return sendImage;
    }
    public void setSendImage(boolean sendImage) {
        this.sendImage = sendImage;
    }
    public Channel getChannel() {
        return channel;
    }
    public void setChannel(Channel channel) {
        this.channel = channel;
    }
    public Role getRole() {
        return role;
    }
    public void setRole(Role role) {
        this.role = role;
    }
}

File: Discord_win/Discord_win/Models/ChannelPermission.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Discord.Models {
    public class ChannelPermission {
        public int ChannelId { get; set; }
        public int RoleId { get; set; }
        public bool ViewMessage { get; set; }
        public bool React { get; set; }
        public bool SendMessage { get; set; }
        public bool SendImage { get; set; }

        public Channel Channel { get; set; }
        public Role Role { get; set; }
    }
}
